### PR TITLE
[WASI-NN] Update ggml

### PR DESCRIPTION
**Pull request by dm4**
- Update ggml library (#2802)
- Use gguf model for tests
- Enable `LLAMA_BLAS` by default
- Pass model filename to ggml backend for preloaded model

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: plugins/wasi_nn/ggml.cpp [hunk 1]
```diff
@@ -112,11 +124,15 @@ Expect<ErrNo> compute(WasiNNEnvironment &Env, uint32_t ContextId) noexcept {
     return ErrNo::InvalidArgument;
   }
 
-  // Output start from prompt.
-  for (auto Id : CxtRef.LlamaInputs) {
-    CxtRef.LlamaOutputs += llama_token_to_str(GraphRef.LlamaContext, Id);
+  // Use env LLAMA_LOG=1 to enable llama log.
+  const char *LlamaLogEnv = std::getenv("LLAMA_LOG");
+  if (LlamaLogEnv != nullptr) {
+    spdlog::info("llama_system_info: {}"sv, llama_print_system_info());
```
Comment (hydai) [inline plugins/wasi_nn/ggml.cpp:130]: Add prefix: `[WASI-NN] GGML backend:`

File: plugins/wasi_nn/ggml.cpp [hunk 2]
```diff
@@ -145,19 +161,24 @@ Expect<ErrNo> compute(WasiNNEnvironment &Env, uint32_t ContextId) noexcept {
...
     CxtRef.LlamaOutputs +=
-        llama_token_to_str(GraphRef.LlamaContext, NewTokenId);
+        llama_token_to_piece(GraphRef.LlamaContext, NewTokenId);
 
     // Push this new token for next evaluation.
     CxtRef.LlamaInputs.push_back(NewTokenId);
   }
 
+  if (LlamaLogEnv != nullptr) {
+    spdlog::info("llama_get_kv_cache_token_count {}"sv,
```
Comment (hydai) [inline plugins/wasi_nn/ggml.cpp:177]: ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]
LGTM

**Pull request merged**